lib/util/include/hse_util/compiler.h

```diff
@@ -8,95 +8,114 @@
 
 #include "build_config.h"
 
+/* clang-format off */
+
 /* Ubuntu 18.04 uses glibc 2.27, and threads.h only exists in glibc >= 2.28 */
 #if defined(__has_include) && !__has_include(<threads.h>)
-#define thread_local _Thread_local
+#define thread_local            _Thread_local
 #else
 #include <threads.h>
 #endif
 
 #if HSE_MOCKING
 #define hse_static
 #else
-#define hse_static static
+#define hse_static              static
 #endif
 
 /* Optimization barrier */
 /* The "volatile" is due to gcc bugs */
-#define barrier() asm volatile("" : : : "memory")
+#define barrier()               asm volatile("" : : : "memory")
+
+#define HSE_LIKELY(_expr)       __builtin_expect(!!(_expr), 1)
+#define HSE_UNLIKELY(_expr)     __builtin_expect(!!(_expr), 0)
 
-#define HSE_LIKELY(x)   __builtin_expect(!!(x), 1)
```

Comment:
These changes feel superfluous...

Reply:
Look at it again with ignoring whitespace and notice the use of the leading and trailing underscores on the attributes.
I added the lines between the #endif #ifdef because there's really no reason it should be so unreadable...

Reply:
Then you should probably submit a patch to the GCC documentation :). They don't even do that.

Reply:
You should also update the attribute checks in lib/meson.build

Reply:
Well, a simple egrep of common attributes in /usr/include shows 783 that do and 863 that don't, where it looks like most of the ones that don't come from 3rd-party packages..
I'll take a look at meson, didn't realize you were generating attributes there...

Reply:
Not generating attributes, but checking if the compiler supports said attributes. Might just need some small tweaks.

Reply:
Oh, that cc.compiles thing is cool, reminds me of autoconf!

Reply:
You may also want to add the double underscores to include/hse/hse.h

Reply:
It looks ok, since the programs are so tiny I don't think there's any possibility of renaming..  But adding underscores should do no harm...   But next PR so I don't have to retest :-)